metaseq/data/filter_dataset.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import numpy as np
+import pandas as pd
+import json
+
+from metaseq.file_io import PathManager
+
+from metaseq.data import data_utils
+from . import BaseWrapperDataset
+
+class FilterDataset(BaseWrapperDataset):
+    """Filters dataset by excluding examples that are easy/hard to learn.
+    
+    Hard/easy to learn are defined by a metric between 0.0 and 1.0, with
```

Comment:
I took this part from the beyond neural scaling law paper - in reality, we might want to include a mix of "hard" and "easy" points (or maybe metrics are not able to even cleanly distinguish between the two); we can add more nuanced logic here to include datapoints in ranges of value instead of just going from 0.0 --> 1.0